.github/workflows/course-schedule.yml

```diff
@@ -0,0 +1,25 @@
+name: "Course Schedule Updates"
+on:
+  - pull_request_target
```

Comment:
Note that the schedule in this PR won't update, since this directive indicates that the action should run based on configuration in the target branch -- and it hasn't been merged yet.

Reply:
We should probably think about the implications of this!

Reply:
We use

```yaml
on:
  pull_request:
```

in the `build.yml`. The difference is that `pull_request` run the action in the context of the (future) merge commit, whereas `pull_request_target` runs at the [base of the branch](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target).

Are you running it at the base to enable the read/write behavior of the `GITHUB_TOKEN`?

Reply:
I don't think it's necessary to split it. Pull requests from new contributors have to be approved before they run — precisely to protect against injection of malicious build steps.

So it should be fine to add the necessary `permissions:` key to a regular build: we check that the job doesn't change in strange ways before we approve PR builds for new contributors. 

That's the theory, as far as I understand it :smile: The real problem is if this understanding is wrong and/or if it's not shared with others who have write access to the repository.